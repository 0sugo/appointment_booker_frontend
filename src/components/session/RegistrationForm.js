import React, { useState } from 'react';
import { Link } from 'react-router-dom';

const RegistrationForm = () => {
  const [formData, setFormData] = useState({
    name: '',
    email: '',
    password: '',
    password_confirmation: '',
  });

  const [registrationSuccess, setRegistrationSuccess] = useState(false);
  const [registrationFailure, setRegistrationFailure] = useState(false);
  const [errorData, setErrorData] = useState(null);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((prevData) => ({
      ...prevData,
      [name]: value,
    }));
  };

  const url = 'http://127.0.0.1:4000/api/v1/users';
  const handleSubmit = async (e) => {
    e.preventDefault();

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ user: formData }),
      });

      if (response.ok) {
        setRegistrationSuccess(true);
      } else {
        const data = await response.json();
        setErrorData(data.errors);
        setRegistrationFailure(true);
      }
    } catch (error) {
      throw new Error(error);
    }
  };

  return (
    <div>
      <h2>Register</h2>
      {registrationSuccess ? (
        <div>
          <p>Registration successful! You can now log in.</p>
        </div>
      ) : (
        <form onSubmit={handleSubmit}>
          <label htmlFor="name">
            Name:
            <input
              id="name"
              type="text"
              name="name"
              value={formData.name}
              onChange={handleChange}
            />
          </label>
          {registrationFailure && errorData && errorData.name && (
            <p>
              Name
              {errorData.name[0]}
            </p>
          )}
          <br />
          <label htmlFor="email">
            Email:
            <input
              id="email"
              type="email"
              name="email"
              value={formData.email}
              onChange={handleChange}
            />
          </label>
          {registrationFailure && errorData && errorData.email && (
            <p>
              Email
              {errorData.email[0]}
            </p>
          )}
          <br />
          <label htmlFor="password">
            Password:
            <input
              id="password"
              type="password"
              name="password"
              value={formData.password}
              onChange={handleChange}
            />
          </label>

          {registrationFailure && errorData && errorData.password && (
            <p>
              Password
              {errorData.password[0]}
            </p>
          )}
          <br />
          <label htmlFor="confirm_password">
            Confirm Password:
            <input
              id="confirm_password"
              type="password"
              name="password_confirmation"
              value={formData.password_confirmation}
              onChange={handleChange}
            />
          </label>
          {registrationFailure && errorData && errorData.password_confirmation && (
            <p>
              Password
              {errorData.password_confirmation[0]}
            </p>
          )}
          <br />
          <button type="submit">Register</button>
        </form>
      )}
      <Link to="/login">Login</Link>
    </div>
  );
};

export default RegistrationForm;
