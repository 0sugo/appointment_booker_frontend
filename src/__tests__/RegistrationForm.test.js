import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import RegistrationForm from '../components/session/RegistrationForm';

describe('Registration form', () => {
  test('renders correctly', () => {
    const regisForm = render(
      <Router>
        <RegistrationForm />
      </Router>,
    );
    expect(regisForm).toMatchSnapshot();
  });
});
