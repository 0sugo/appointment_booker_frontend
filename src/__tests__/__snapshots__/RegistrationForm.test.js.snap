// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration form renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="regis_page"
      >
        <h2>
          Register
        </h2>
        <form>
          <label
            for="name"
          >
            Name:
            <input
              id="name"
              name="name"
              type="text"
              value=""
            />
          </label>
          <br />
          <label
            for="email"
          >
            Email:
            <input
              id="email"
              name="email"
              type="email"
              value=""
            />
          </label>
          <br />
          <label
            for="password"
          >
            Password:
            <input
              id="password"
              name="password"
              type="password"
              value=""
            />
          </label>
          <br />
          <label
            for="confirm_password"
          >
            Confirm Password:
            <input
              id="confirm_password"
              name="password_confirmation"
              type="password"
              value=""
            />
          </label>
          <br />
          <div
            class="reg_btns"
          >
            <button
              class="splash_link login_btn"
              type="submit"
            >
              Register
            </button>
            <a
              class="splash_link"
              href="/login"
            >
              Login
            </a>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="regis_page"
    >
      <h2>
        Register
      </h2>
      <form>
        <label
          for="name"
        >
          Name:
          <input
            id="name"
            name="name"
            type="text"
            value=""
          />
        </label>
        <br />
        <label
          for="email"
        >
          Email:
          <input
            id="email"
            name="email"
            type="email"
            value=""
          />
        </label>
        <br />
        <label
          for="password"
        >
          Password:
          <input
            id="password"
            name="password"
            type="password"
            value=""
          />
        </label>
        <br />
        <label
          for="confirm_password"
        >
          Confirm Password:
          <input
            id="confirm_password"
            name="password_confirmation"
            type="password"
            value=""
          />
        </label>
        <br />
        <div
          class="reg_btns"
        >
          <button
            class="splash_link login_btn"
            type="submit"
          >
            Register
          </button>
          <a
            class="splash_link"
            href="/login"
          >
            Login
          </a>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
